src/fireedge/src/server/routes/api/files/functions.js

```diff
@@ -0,0 +1,439 @@
+ */
+const validateFileWithoutOwner = (file = '') => {
+  const parsedFile = parseFilePath(file)
+
+  return (
+    Array.isArray(parsedFile) &&
+    parsedFile[0] &&
+    checkValidApp(parsedFile[0]) &&
+    !parsedFile[1]
+  )
+}
+
+/**
+ * Check if user is a file owner.
+ *
+ * @param {string} file - filename
+ * @param {number} id - user id
+ * @returns {boolean} - if user is the file owner
+ */
+const validateFileWithOwner = (file = '', id = '') => {
+  const parsedFile = parseFilePath(file)
+
+  return (
+    Array.isArray(parsedFile) &&
+    parsedFile[0] &&
+    checkValidApp(parsedFile[0]) &&
+    parsedFile[1] &&
+    parsedFile[1] === id
+  )
+}
+
+/**
+ * Upload File.
+ *
+ * @param {object} res - response http
+ * @param {Function} next - express stepper
+ * @param {string} params - data response http
+ * @param {object} userData - user of http request
+ * @param {Function} oneConnection - xmlrpc connection
+ */
+const upload = (
+  res = {},
+  next = defaultEmptyFunction,
+  params = {},
+  userData = {},
+  oneConnection = defaultEmptyFunction
+) => {
+  const { app, files, public: publicFile } = params
+  const { id, user, password } = userData
+  if (
+    !(
+      global.paths.CPI &&
+      app &&
+      checkValidApp(app) &&
+      files &&
+      id &&
+      user &&
+      password
+    )
+  ) {
+    res.locals.httpCode = httpBadRequest
+    next()
+  }
+
+  const oneConnect = oneConnection(user, password)
+  checkUserAdmin(
+    oneConnect,
+    id,
+    (admin = false) => {
+      const pathUserData = publicFile && admin ? `${app}` : `${app}${sep}${id}`
+      const pathUser = `${global.paths.CPI}${sep}${pathUserData}`
+      if (!existsSync(pathUser)) {
+        mkdirsSync(pathUser)
+      }
+      let method = ok
+      let message = ''
+      const data = []
+      for (const file of files) {
+        if (file && file.originalname && file.path && file.filename) {
+          const extFile = extname(file.originalname)
+          try {
+            const filenameApi = `${pathUserData}${sep}${file.filename}${extFile}`
+            const filename = `${pathUser}${sep}${file.filename}${extFile}`
+            moveSync(file.path, filename)
+            data.push(filenameApi)
+          } catch (error) {
+            method = internalServerError
+            message = error && error.message
+            break
+          }
+        }
+      }
+      res.locals.httpCode = httpResponse(
+        method,
+        data.length ? data : '',
+        message
+      )
+      next()
+    },
+    () => {
+      res.locals.httpCode = internalServerError
+      next()
+    }
+  )
+}
+
+/**
+ * Get default files for app.
+ *
+ * @param {string} app - app
+ * @param {boolean} multiple - find multiple files
+ * @param {string} defaultFile - default file
+ * @returns {Array | string} - file
+ */
+const getDefaultFilesforApps = (
+  app = '',
+  multiple = false,
+  defaultFile = ''
+) => {
+  let rtn = ''
+  switch (app) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are using less than 3 statements in `switch`. It&#39;s better to use `if-else` instead.